migrator/migrations/m_112_to_m_113_recreate_groups_bucket/migration.go

```diff
@@ -0,0 +1,157 @@
+	}
+
+	return nil
+}
+
+func fetchGroupsBucket(db *bolt.DB) (groupEntries []groupEntry, err error) {
+	err = db.View(func(tx *bolt.Tx) error {
+		bucket := tx.Bucket(bucketName)
+		// We previously checked that the bucket should be available, but still add this check here.
+		if bucket == nil {
+			return nil
+		}
+		return bucket.ForEach(func(k, v []byte) error {
+			groupEntries = append(groupEntries, groupEntry{key: k, value: v})
+			return nil
+		})
+	})
+	if err != nil {
+		return nil, err
+	}
+	return groupEntries, nil
+}
+
+func dropGroupsBucket(db *bolt.DB) error {
+	return db.Update(func(tx *bolt.Tx) error {
+		return tx.DeleteBucket(bucketName)
+	})
+}
+
+func createGroupsBucket(db *bolt.DB, groupEntries []groupEntry) (err error) {
+	err = db.Update(func(tx *bolt.Tx) error {
+		// Explicitly use the CreateBucket here instead of CreateBucketIfNotExists, as we require the bucket to be
+		// previously dropped.
+		bucket, err := tx.CreateBucket(bucketName)
+		if err != nil {
+			return err
+		}
+
+		var putGroupErrs errorhelpers.ErrorList
+		for _, entry := range groupEntries {
+			// By now, we can assume that the key will be a UUID and the value will be the group proto message.
+			// Here, we will check that the key will be a string and can be parsed as a UUID.
+			// If that's the case, the entry is valid, and we will add it to the re-created bucket.
+			// If not, we will log the invalid entry that will be dropped.
+			if !verifyKeyValuePair(entry.key, entry.value) {
+				log.WriteToStderrf("Invalid group entry found in groups bucket (key=%s, value=%s). This entry"+
+					" will be dropped.",
+					entry.key, entry.value)
+				continue
+			}
+
+			if err := bucket.Put(entry.key, entry.value); err != nil {
+				putGroupErrs.AddError(err)
+			}
+		}
+
+		return putGroupErrs.ToError()
+	})
+	return err
+}
+
+func checkGroupBucketExists(db *bolt.DB) (exists bool, err error) {
+	exists = true
+	err = db.View(func(tx *bolt.Tx) error {
+		bucket := tx.Bucket(bucketName)
+		if bucket == nil {
+			exists = false
+		}
+		return nil
+	})
+	return exists, err
+}
+
+const (
+	// Value has been taken from:
+	//	https://github.com/stackrox/stackrox/blob/master/central/group/datastore/validate.go#L13
+	groupIDPrefix = "io.stackrox.authz.group."
+	// Value has been taken from:
+	//	https://github.com/stackrox/stackrox/blob/master/migrator/migrations/m_105_to_m_106_group_id/migration.go#L134
+	groupMigratedIDPrefix = "io.stackrox.authz.group.migrated."
+)
+
+func verifyKeyValuePair(key, value []byte) bool {
+	stringKey := string(key)
+
+	// The key should be a string ID, with a constant prefix and a UUID.
+	if !strings.HasPrefix(stringKey, groupIDPrefix) && !strings.HasPrefix(stringKey, groupMigratedIDPrefix) {
+		return false
+	}
+
+	// The value should be a storage.Group with ID set.
+	var group storage.Group
+	if err := proto.Unmarshal(value, &group); err != nil {
+		return false
+	}
+	return group.GetProps().GetId() != ""
```

Comment:
Also curious why we give up on groups without props id here

Reply:
So, no group should exist after the migration we had beforehand which doesn't have an ID associated with it.
This is just to ensure we have no "invalid" group properties in the store (if you were to try to delete this specific group, you wouldn't be able to at the moment).